parasoft-findings-teamcity-agent/src/main/java/com/parasoft/findings/teamcity/agent/LocalSettingsHelper.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2023 Parasoft Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.parasoft.findings.teamcity.agent;
+
+import com.parasoft.xtest.common.text.UString;
+import jetbrains.buildServer.agent.AgentRunningBuild;
+import org.apache.commons.io.IOUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class LocalSettingsHelper {
+    private static final String validUrlPattern = "(\\w+)://([\\w.]+)/(\\S*)";
+    private static final Logger LOG = Logger.getLogger
+            (LocalSettingsHelper.class.getName()); // logs into ./buildAgent/logs/wrapper.log
+
+    private AgentRunningBuild _build;
+
+    public LocalSettingsHelper(AgentRunningBuild _build) {
+        this._build = _build;
+    }
+
+    public Properties loadLocalSettings(File checkoutDir, String settingsPath) {
+        if (UString.isEmpty(settingsPath)) {
+            return new Properties();
+        }
+        File localSettingsFile = new File(settingsPath);
+        if (!localSettingsFile.exists()) {
```

Comment:
Let's include a condition check for the isAbsolute() method to determine if the file path is absolute or not. This may not significantly affect the logic, but it can make the code easier to read without a comment.

Reply:
So the logic could be:
if (absolute and exist) -> use settings path
else -> append checkout directory